vic/drivers/classic/src/alloc_veg_hist.c

```diff
@@ -38,29 +38,27 @@ alloc_veg_hist(int                nrecs,
 {
     int i, j;
 
-    (*veg_hist) = (veg_hist_struct **) calloc(nrecs, sizeof(veg_hist_struct *));
+    (*veg_hist) = calloc(nrecs, sizeof(*(*veg_hist)));
     if ((*veg_hist) == NULL) {
         log_err("Memory allocation error in alloc_veg_hist().");
     }
 
     for (i = 0; i < nrecs; i++) {
-        (*veg_hist)[i] = (veg_hist_struct *) calloc(nveg,
-                                                    sizeof(veg_hist_struct));
+        (*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i]));
```

Comment:
Dang, did I write the code that way?  Sorry about that.  I must have copied
it from some other function.  It's really awkward.

On Thu, Aug 13, 2015 at 10:07 PM, Joe Hamman <notifications@github.com>
wrote:

> In vic/drivers/classic/src/alloc_veg_hist.c
> <https://github.com/UW-Hydro/VIC/pull/257#discussion_r37049978>:
>
> >      if ((*veg_hist) == NULL) {
> >          log_err("Memory allocation error in alloc_veg_hist().");
> >      }
> >
> >      for (i = 0; i < nrecs; i++) {
> > -        (*veg_hist)[i] = (veg_hist_struct *) calloc(nveg,
> > -                                                    sizeof(veg_hist_struct));
> > +        (*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i]));
>
> I'll leave the style decision up to you since I haven't crossed this
> bridge before.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/UW-Hydro/VIC/pull/257/files#r37049978>.
>